src/blockchain/merkle-patricia.js

```diff
@@ -179,25 +190,18 @@ function precheckProof () {
   })
 }
 
-function serializeProofPathType (type, buffer) {
-  Bool.serialize(type.isTerminal, buffer, buffer.length)
-  Hash.serialize(type.hexKey, buffer, buffer.length)
-  Uint8.serialize(type.lengthByte, buffer, buffer.length)
-  return buffer
-}
-
 function serializeBranchNode (leftHash, rightHash, leftPath, rightPath) {
-  let buffer = []
+  const buffer = [MAP_BRANCH_PREFIX]
   Hash.serialize(leftHash, buffer, buffer.length)
   Hash.serialize(rightHash, buffer, buffer.length)
-  serializeProofPathType(leftPath, buffer)
-  serializeProofPathType(rightPath, buffer)
+  leftPath.serialize(buffer)
+  rightPath.serialize(buffer)
   return buffer
 }
 
 function serializeIsolatedNode (path, hash) {
-  let buffer = []
-  serializeProofPathType(path, buffer)
+  // `1, ...path.key, 0` is the old `ProofPath` serialization for terminal paths.
```

Comment:
@aleksuss Is there a particular reason why `ProofPath` is serialized in this way here? I'd think using the new serialization would be more logical.

Reply:
@slowli Do you mean why do not use `protobuf` here instead ? Or format is confusing you ?

Reply:
I mean why not use `LEB128(key_len) || key_bytes`, as when serializing branches with 2 children.